### PR TITLE
Remove dependency on Microsoft.VisualStudio.RemoteControl assembly from Features

### PR DESCRIPTION
**Pull request by tmat**
Features assembly can only reference OSS libraries.

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi PTAL

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.RemoteControlService.cs [hunk 1]
```diff
@@ -2,24 +2,9 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
+namespace Microsoft.CodeAnalysis.SymbolSearch;
 
-using Microsoft.VisualStudio.RemoteControl;
-
-namespace Microsoft.CodeAnalysis.SymbolSearch
+internal partial class SymbolSearchUpdateEngine
```
Comment (ryzngard) [inline src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.RemoteControlService.cs:7]: can you add a comment about why we need a blank internal partial here?

**Review by ryzngard — APPROVED** [review]
LGTM for layering

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.RemoteControlService.cs (hunk 1 above)
Comment (tmat) [reply src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.RemoteControlService.cs]: Should have been deleted.

**Comment by jasonmalinowski** [comment]
Is this used in VS for Mac? It obviously shouldn't be in Features, but is the VS for Windows too high of a layer?

**Comment by tmat** [comment]
It's Windows only. The impl is linked to both VS and ServiceHub.Remote projects. Conveniently we already have a few similarly linked storage related types. We can consider moving all of them to LSP layer, if needed.

**Pull request merged**